test/Microsoft.Framework.Internal.Test/BufferEntryCollectionTest.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Framework.Internal
+{
+    public class BufferEntryCollectionTest
+    {
+        [Fact]
+        public void Add_AddsBufferEntries()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+            var inner = new BufferEntryCollection();
+
+            // Act
+            collection.Add("Hello");
+            collection.Add(new[] { 'a', 'b', 'c' }, 1, 2);
+            collection.Add(inner);
+
+            // Assert
+            Assert.Equal(new[] { "Hello", "bc" }, collection);
+            Assert.Same(inner.BufferEntries, collection.BufferEntries[2]);
+        }
+
+        [Fact]
+        public void AddChar_ThrowsIfIndexIsOutOfBounds()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+
+            // Act and Assert
+            var ex = Assert.Throws<ArgumentOutOfRangeException>(
+                () => collection.Add(new[] { 'a', 'b', 'c' }, -1, 2));
+            Assert.Equal("index", ex.ParamName);
+            Assert.Equal(
+                "Specified argument was out of the range of valid values.\r\nParameter name: index",
+                ex.Message);
+        }
+
+        [Fact]
+        public void AddChar_ThrowsIfCountWouldCauseOutOfBoundReads()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+
+            // Act and Assert
+            var ex = Assert.Throws<ArgumentOutOfRangeException>(
+                () => collection.Add(new[] { 'a', 'b', 'c' }, 1, 3));
+            Assert.Equal("count", ex.ParamName);
+            Assert.Equal(
+                "Specified argument was out of the range of valid values.\r\nParameter name: count",
+                ex.Message);
+        }
+
+        public static TheoryData AddWithChar_RepresentsStringsAsChunkedEntriesData
+        {
+            get
+            {
+                return new TheoryData<char[], int, int, IList<object>> {
```

Comment:
a comment would help; should explain what the values in this set mean since reader hasn't seen test method yet